packages/storage/src/implementation/requests.ts

```diff
@@ -279,6 +279,41 @@ export function metadataForUpload_(
   return metadataClone;
 }
 
+/**
+ * Prepare RequestInfo for non-resumable uploads as Content-Type: octet-stream.
+ */
+export function simpleUpload(
+  service: StorageService,
+  location: Location,
+  mappings: MetadataUtils.Mappings,
+  blob: FbsBlob,
+  metadata?: Metadata | null
+): RequestInfo<Metadata> {
+  const urlPart = location.bucketOnlyServerUrl();
+  const headers: { [prop: string]: string } = {
+    'Content-Type': 'application/octet-stream'
+  };
+
+  const metadata_ = metadataForUpload_(location, blob, metadata);
+  const urlParams: UrlParams = { name: metadata_['fullPath']! };
+  const url = UrlUtils.makeUrl(urlPart);
```

Comment:
Did this in files touched by this PR but I would like to go through and do another PR for files not touched by this PR.

Reply:
FWIW, I have gone back on forth on this. Is this actually true in builds that are bundled? I would assume that only the import that make it into the final bundle matter, and most of these internal imports should be removed. This is just a guess though, and in general I am avoiding namespace imports.

Reply:
Rollup will remove unused functions even when using namespace imports. So to your point, it doesn't matter whether or not namespace imports are used in bundled builds. I still think we should use namespace imports as the best practice because they are likely to work well with any build tool we might use. 

index.ts
```
import * as impl from './impl';

console.log(impl.foo());
```

impl.ts
```
export function foo() {}

// removed in the bundle
export function bar() {}
```

bundle.js produced by rollup
```
function foo() {
}

// import * as firebase from 'firebase/app';
console.log(foo());

```